solr/core/src/java/org/apache/solr/servlet/SolrRequestParsers.java

```diff
@@ -549,7 +546,9 @@ public InputStream getStream() throws IOException {
     public SolrParams parseParamsAndFillStreams( 
         final HttpServletRequest req, ArrayList<ContentStream> streams ) throws Exception
     {
-      streams.add( new HttpRequestContentStream( req ) );
+      if (!req.getMethod().equals("GET")) { // or other conditions? Check headers?
+        streams.add(new HttpRequestContentStream(req));
+      }
```

Comment:
@noblepaul  when doing a GET request for V2, I found it to be a problem that we add this stream that is empty, resulting in an exception at some point about a blank Content-Type (as there is no content in the first place).  StandardRequestParser.parseParamsAndFillStreams checks !isPost and isV2 and then calls this raw impl.
I could probably be more thorough here in detecting the presence of a stream.  Just because it's not a GET doesn't mean there is a stream!  Weirdly, req.getInputStream can return non-null for a GET which I observed in a test I added in this PR.

Reply:
I should enhance this condition to look for `Content-Length` or `Transfer-Encoding` headers, which the HTTP RFC says is how you know if there is a message body.

Reply:
Please enhance this and I shall review it

Reply:
See #134 -- I split this part out to its own.  I requested your review there already.